Pyrado/pyrado/utils/experiments.py

```diff
@@ -67,24 +74,176 @@ def load_experiment(
+        env = pyrado.load("env.pkl", ex_dir)
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy, verbose=True)
+
+    elif algo.name in ["dql", "sac"]:
+        # Environment
+        env = pyrado.load("env.pkl", ex_dir)
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy, verbose=True)
+        # Target value functions
+        if algo.name == "dql":
+            extra["qfcn_target"] = pyrado.load("qfcn_target.pt", ex_dir, obj=algo.qfcn_targ, verbose=True)
+        elif algo.name == "sac":
+            extra["qfcn_target1"] = pyrado.load("qfcn_target1.pt", ex_dir, obj=algo.qfcn_targ_1, verbose=True)
+            extra["qfcn_target2"] = pyrado.load("qfcn_target2.pt", ex_dir, obj=algo.qfcn_targ_2, verbose=True)
+        else:
+            raise NotImplementedError
+
+    elif algo.name == "svpg":
+        # Environment
+        env = pyrado.load("env.pkl", ex_dir)
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy, verbose=True)
+        # Extra (particles)
+        for idx, p in enumerate(algo.particles):
+            extra[f"particle{idx}"] = pyrado.load(f"particle_{idx}.pt", ex_dir, obj=algo.particles[idx], verbose=True)
+
+    elif algo.name == "tspred":
+        # Dataset
+        extra["dataset"] = to.load(osp.join(ex_dir, "dataset.pt"))
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy, verbose=True)
+
+    elif algo.name == "sprl":
+        # Environment
+        env = pyrado.load("env.pkl", ex_dir)
+        print_cbt(f"Loaded {osp.join(ex_dir, 'env.pkl')}.", "g")
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy)
+        print_cbt(f"Loaded {osp.join(ex_dir, f'{args.policy_name}.pt')}", "g")
+        # Extra (value function)
+        if isinstance(algo._subroutine, ActorCritic):
+            extra["vfcn"] = pyrado.load(f"{args.vfcn_name}.pt", ex_dir, obj=algo._subroutine.critic.vfcn, verbose=True)
+
+    elif algo.name == "pddr":
+        # Environment
+        env = pyrado.load("env.pkl", ex_dir)
+        # Policy
+        policy = pyrado.load(f"{args.policy_name}.pt", ex_dir, obj=algo.policy, verbose=True)
+        # Teachers
+        extra["teacher_policies"] = algo.teacher_policies
+        extra["teacher_envs"] = algo.teacher_envs
+        extra["teacher_expl_strats"] = algo.teacher_expl_strats
+        extra["teacher_critics"] = algo.teacher_critics
+        extra["teacher_ex_dirs"] = algo.teacher_ex_dirs
+
+    else:
+        raise pyrado.TypeErr(msg="No matching algorithm name found during loading the experiment!")
 
     # Check if the return types are correct. They can be None, too.
-    if env is not None and not isinstance(env, (Env, EnvWrapper)):
-        raise pyrado.TypeErr(given=env, expected_type=[Env, EnvWrapper])
+    if env is not None and not isinstance(env, (SimEnv, EnvWrapper)):
```

Comment:
Sure about this change? Could this lead to conflicts with real envs?

Reply:
Merging error…